lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
I think you misunderstood me. I said that `Client::serialize` is public and shouldn't be because there is no reason to call this method outside of the `Client` class itself to serialize values sent to the server. And while that class doesn't implement `SerializerInterface`, then we are free to make the method protected